sql/connect/server/src/main/scala/org/apache/spark/sql/connect/execution/SparkConnectPlanExecution.scala

```diff
@@ -144,6 +144,14 @@ private[execution] class SparkConnectPlanExecution(executeHolder: ExecuteHolder)
           sendBatch(bytes, count, offset)
           offset += count
         }
+      case cle @ (_: CollectLimitExec | _: CollectTailExec) =>
+        val rows = cle.executeCollect()
```

Comment:
I'd be nice if we could do this in a more streaming fashion. That is the whole point of connect's custom collection code path.

Reply:
Thanks for your review, I will rewrite it in streaming fashion.